src/Tests/dotnet-publish.Tests/GivenDotnetPublishPublishesProjects.cs

```
@@ -137,6 +141,89 @@ public void ItPublishesSelfContainedWithRid(string args)
+        {
+            bool selfContained = !publishSelfContained;
+            bool resultShouldBeSelfContained = publishSelfContained && !selfContainedIsGlobal;
+
+            string targetFramework = ToolsetInfo.CurrentTargetFramework;
+            var testProject = new TestProject("MainProject")
+            {
+                TargetFrameworks = targetFramework,
+                IsExe = true
+            };
+
+            testProject.RecordProperties("SelfContained");
+            if (!publishSelfContainedIsGlobal)
+                testProject.AdditionalProperties["PublishSelfContained"] = publishSelfContained.ToString();
+            if (!selfContainedIsGlobal)
+                testProject.AdditionalProperties["SelfContained"] = selfContained.ToString();
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject, identifier: $"PSC-OVERRIDES-{publishSelfContained}-{selfContainedIsGlobal}-{publishSelfContainedIsGlobal}");
+            var publishCommand = new DotnetCommand(Log);
+            List<string> args = new List<string>
+            {
+                "publish",
+                selfContainedIsGlobal ? $"/p:SelfContained={selfContained}" : "",
+                publishSelfContainedIsGlobal ? $"/p:PublishSelfContained={publishSelfContained}" : "",
+            };
+
+            publishCommand
+                .WithWorkingDirectory(Path.Combine(testAsset.Path, "MainProject"))
+                .Execute(args.ToArray())
+                .Should()
+                .Pass();
+
+            var publishedDirectory = testAsset.TestRoot;
+            publishedDirectory = Path.Combine(publishedDirectory, "MainProject", "obj", "Debug", targetFramework);
+            if (resultShouldBeSelfContained)
+                publishedDirectory = Directory.GetDirectories(publishedDirectory).FirstOrDefault(); // get the rid in the directory
```

If there is more than one RID directory, then it would be better to throw an exception here rather than to return null and fail later.

```suggestion
                publishedDirectory = Directory.GetDirectories(publishedDirectory).First(); // get the rid in the directory
```

---

Good point, thanks. With the below change this won't be needed.